content/docs/tutorials/handling-errors.mdx

```diff
@@ -0,0 +1,272 @@
+Just like with query errors, there are dozens of possible causes of a transaction submission failure and they all return their own internal result codes. Horizon attempts to wrangle the asynchronous nature of the beast that is Stellar-Core into a synchronous flow more familiar to application developers. This requires making assumptions and imposing restrictions,  and though it's a simpler model there are still caveats to be aware of.
+
+Here's a common failure: using the wrong sequence number for an account. The following transaction envelope (examine it in the [Laboratory](https://laboratory.stellar.org/#xdr-viewer?input=AAAAAgAAAABjsEXePrFn%2BdH3tNaTkGo3kojf2iecJqU3g84%2F95XLqgAAAGQAAAAAbG1hbwAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAA50aGlzIGlzIGEgdGVzdAAAAAAAAQAAAAEAAAAAY7BF3j6xZ%2FnR97TWk5BqN5KI39onnCalN4POP%2FeVy6oAAAAKAAAAB21lc3NhZ2UAAAAAAQAAAA1IZWxsbywgd29ybGQhAAAAAAAAAAAAAAH3lcuqAAAAQNZ%2F9lkhKMw0Rbiwmw%2BdF3X1DldjT0qPdiCuEPpf5rErYDoXBkr0HlQdAfK1ptfLwK7kMELQfXCfPUEYVuJd4gQ%3D&type=TransactionEnvelope&network=test)) is a simple transaction to add some [managed data](../start/list-of-operations#manage-data) to the account:
+
+<CodeExample>
+
+```bash
+curl -s --data "tx=AAAAAgAAAABjsEXePrFn+dH3tNaTkGo3kojf2iecJqU3g84/95XLqgAAAGQAAAAAbG1hbwAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAA50aGlzIGlzIGEgdGVzdAAAAAAAAQAAAAEAAAAAY7BF3j6xZ/nR97TWk5BqN5KI39onnCalN4POP/eVy6oAAAAKAAAAB21lc3NhZ2UAAAAAAQAAAA1IZWxsbywgd29ybGQhAAAAAAAAAAAAAAH3lcuqAAAAQNZ/9lkhKMw0Rbiwmw+dF3X1DldjT0qPdiCuEPpf5rErYDoXBkr0HlQdAfK1ptfLwK7kMELQfXCfPUEYVuJd4gQ=" https://horizon-testnet.stellar.org | jq
+```
+
+</CodeExample>
+
+<CodeExample>
+
+```json
+{
+  "type": "https://stellar.org/horizon-errors/transaction_failed",
+  "title": "Transaction Failed",
+  "status": 400,
+  "detail": "The transaction failed when submitted to the stellar network. The `extras.result_codes` field on this response contains further details.  Descriptions of each code can be found at: https://www.stellar.org/developers/guides/concepts/list-of-operations.html",
+  "extras": {
+    "envelope_xdr": "AAAAAgAAAABjsEXePrFn+dH3tNaTkGo3kojf2iecJqU3g84/95XLqgAAAGQAAAAAbG1hbwAAAAEAAAAAAAAAAAAAAAAAAAAAAAAAAQAAAA50aGlzIGlzIGEgdGVzdAAAAAAAAQAAAAEAAAAAY7BF3j6xZ/nR97TWk5BqN5KI39onnCalN4POP/eVy6oAAAAKAAAAB21lc3NhZ2UAAAAAAQAAAA1IZWxsbywgd29ybGQhAAAAAAAAAAAAAAH3lcuqAAAAQNZ/9lkhKMw0Rbiwmw+dF3X1DldjT0qPdiCuEPpf5rErYDoXBkr0HlQdAfK1ptfLwK7kMELQfXCfPUEYVuJd4gQ=",
+    "result_codes": {
+      "transaction": "tx_bad_seq"
+    },
+    "result_xdr": "AAAAAAAAAAD////7AAAAAA=="
+  }
+}
+```
+
+</CodeExample>
+
+
+## Invalid Sequences
+These errors typically occur when you have an outdated view of an account due to other transactions happening outside of your worldview. This could be because the account is used on multiple devices, you have concurrent submissions happening, or a number of other reasons.
+
+Thankfully, the solution is relatively simple: retrieve the account details and try again with an updated sequence number.
```

Comment:
I think it's worth noting here that updating sequence number to the latest value can sometime cause a double spend. Say you have two submitters running and they both try to send the same payment tx with the same seqnum. One of them will succeed, the other will get bad sequence error and will try to submit exactly the same tx again. If it succeeds this time the payment will be sent twice.